keras/src/layers/layer.py

```diff
@@ -1280,9 +1280,12 @@ def _untrack_variable(self, variable):
             self._tracker.lock()
         self._post_untrack_variable(variable)
 
-    def add_metric(self):
+    def add_metric(self, *args, **kwargs):
         # Permanently disabled
-        raise NotImplementedError
+        raise NotImplementedError(
+            "Layer `add_metric()` method is deprecated"
+            " add your metric in `compile(metrics=[...]).`"
```

Comment:
I would say `Model.compile(metrics=[...]).` since this is the `Layer` class.

Reply:
Good call, switched!